

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Metagenomic screening of shotgun data &mdash; Paleogenomics 0.0.1 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript">
          var DOCUMENTATION_OPTIONS = {
              URL_ROOT:'./',
              VERSION:'0.0.1',
              LANGUAGE:'None',
              COLLAPSE_INDEX:false,
              FILE_SUFFIX:'.html',
              HAS_SOURCE:  true,
              SOURCELINK_SUFFIX: '.txt'
          };
      </script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> Paleogenomics
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul>
<li class="toctree-l1"><a class="reference internal" href="1_ListTools.html">1. List of Tools</a></li>
<li class="toctree-l1"><a class="reference internal" href="2_ReadsFiltering.html">2. Quality filtering of reads</a></li>
<li class="toctree-l1"><a class="reference internal" href="3_Metagenomics_v2.html">3. Metagenomic screening of shotgun data</a></li>
<li class="toctree-l1"><a class="reference internal" href="4_ReadsMapping_v2.html">4. Alignment of reads to a reference genome</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Paleogenomics</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Metagenomic screening of shotgun data</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/Metagenomics.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="metagenomic-screening-of-shotgun-data">
<h1>Metagenomic screening of shotgun data<a class="headerlink" href="#metagenomic-screening-of-shotgun-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="minikraken">
<h2>Minikraken<a class="headerlink" href="#minikraken" title="Permalink to this headline">¶</a></h2>
<div class="section" id="taxonomic-assignation-with-the-pre-built-minikraken-database">
<h3>Taxonomic assignation with the pre-built Minikraken database<a class="headerlink" href="#taxonomic-assignation-with-the-pre-built-minikraken-database" title="Permalink to this headline">¶</a></h3>
<p>In this hands-on session we will use <em>Kraken</em> (<a class="reference external" href="http://ccb.jhu.edu/software/kraken/">http://ccb.jhu.edu/software/kraken/</a>) to screen the metagenomic content of a DNA extract after shotgun sequencing. When analyzing a metagenomics sample using a Kraken database the primary source of false positive hits is low-complexity sequences in the genomes themselves (e.g., a string of 31 or more consecutive A’s). These can largely be eliminated by first running the ‘dust’ program on all genomes and then building the database from these ‘dusted’ genomes.
We will first use a pre-built 8 GB Kraken database (Minikraken) constructed from complete dustmasked bacterial, archaeal, and viral genomes in RefSeq (as of October 2017), and directly run the taxonomic assignation of the reads in our sample with the <cite>kraken</cite> command.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span> <span class="o">--</span><span class="n">db</span> <span class="n">Minikraken</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">fastq</span><span class="o">-</span><span class="nb">input</span> <span class="n">filename</span><span class="o">.</span><span class="n">gz</span> <span class="o">--</span><span class="n">gzip</span><span class="o">-</span><span class="n">compressed</span> <span class="o">--</span><span class="n">output</span> <span class="n">filename</span><span class="o">.</span><span class="n">kraken</span>
</pre></div>
</div>
<p>Some of the options available in Kraken:</p>
<table border="1" class="docutils">
<colgroup>
<col width="25%" />
<col width="75%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Function</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><em>–db</em> &lt;string&gt;</td>
<td>Path to the folder (database name) containing the database files.</td>
</tr>
<tr class="row-odd"><td><em>–output</em> &lt;string&gt;</td>
<td>Print output to filename.</td>
</tr>
<tr class="row-even"><td><em>–threads</em> &lt;integer&gt;</td>
<td>Number of threads (only when multiple cores are used).</td>
</tr>
<tr class="row-odd"><td><em>–fasta-input</em></td>
<td>Input is FASTA format.</td>
</tr>
<tr class="row-even"><td><em>–fastq-input</em></td>
<td>Input is FASTQ format.</td>
</tr>
<tr class="row-odd"><td><em>–gzip-compressed</em></td>
<td>Input is gzip compressed.</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="create-report-files">
<h3>Create report files<a class="headerlink" href="#create-report-files" title="Permalink to this headline">¶</a></h3>
<p>Once the taxonomic assignation is done, we create from the Kraken output file a report of the analysis by running the <cite>kraken-report</cite> script. Note that the database used must be the same as the one used to generate the output file in the command above. The output file is a tab-delimited file with the following fields, from left to right:</p>
<blockquote>
<div><ol class="arabic simple">
<li>Percentage of reads covered by the clade rooted at this taxon</li>
<li>Number of reads covered by the clade rooted at this taxon</li>
<li>Number of reads assigned directly to this taxon</li>
<li>A rank code, indicating (U)nclassified, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. All other ranks are simply ‘-‘.</li>
<li>NCBI taxonomy ID</li>
<li>Indented scientific name</li>
</ol>
</div></blockquote>
<p>Notice that we will have to redirect the output to a file with ‘&gt;’.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">report</span> <span class="o">--</span><span class="n">db</span> <span class="n">Minikraken</span> <span class="n">filename</span><span class="o">.</span><span class="n">kraken</span> <span class="o">&gt;</span> <span class="n">filename</span><span class="o">.</span><span class="n">kraken</span><span class="o">.</span><span class="n">report</span>
</pre></div>
</div>
<ul>
<li><p class="first"><em>Note</em>: We can use a <em>for</em> loop to make the taxonomic assignation and create the report file for multiple samples. Notice the assignation of variables <em>filename</em> and <em>fname</em> to return output files named after the sample.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for i in *.fastq
do
  filename=$(basename &quot;$i&quot;)
  fname=&quot;${filename%.fastq}&quot;
  kraken --db Minikraken --threads 4 --fastq-input $i --output /${fname}.kraken
  kraken-report --db Minikraken ${fname}.kraken &gt; ${fname}.kraken.report
done
</pre></div>
</div>
</li>
</ul>
</div>
<div class="section" id="visualization-of-data-with-krona">
<h3>Visualization of data with Krona<a class="headerlink" href="#visualization-of-data-with-krona" title="Permalink to this headline">¶</a></h3>
<p>Finally, we can visualize the results of the Kraken analysis with <em>Krona</em> (<a class="reference external" href="https://github.com/marbl/Krona/wiki">https://github.com/marbl/Krona/wiki</a>), which disaplys hierarchical data (like taxonomic assignation) in multi-layerd pie charts. The interactive charts created by Krona are in the _html_ format and can be viewed with any web browser.
We will convert the kraken output in html format using the program <cite>ktImportTaxonomy</cite>, which parses the information relative to the query ID and the taxonomy ID.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ktImportTaxonomy</span> <span class="o">-</span><span class="n">q</span> <span class="mi">2</span> <span class="o">-</span><span class="n">t</span> <span class="mi">3</span> <span class="n">filename</span><span class="o">.</span><span class="n">kraken</span> <span class="o">-</span><span class="n">o</span> <span class="n">filename</span><span class="o">.</span><span class="n">kraken</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>Some of the options available in ktImportTaxonomy:</p>
<table border="1" class="docutils">
<colgroup>
<col width="24%" />
<col width="76%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Function</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><em>-q</em> &lt;integer&gt;</td>
<td>Column of input files to use as query ID.</td>
</tr>
<tr class="row-odd"><td><em>-t</em> &lt;integer&gt;</td>
<td>Column of input files to use as taxonomy ID.</td>
</tr>
<tr class="row-even"><td><em>-o</em> &lt;string&gt;</td>
<td>Output file name.</td>
</tr>
</tbody>
</table>
<ul>
<li><p class="first"><em>Note</em>: if you want to analyze multiple kraken files from various samples you view the results in one single html file running _ktImportTaxonomy_ as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ktImportTaxonomy</span> <span class="o">-</span><span class="n">q</span> <span class="mi">2</span> <span class="o">-</span><span class="n">t</span> <span class="mi">3</span> <span class="n">filename_1</span><span class="o">.</span><span class="n">kraken</span> <span class="n">filename_2</span><span class="o">.</span><span class="n">kraken</span> <span class="o">...</span> <span class="n">filename_n</span><span class="o">.</span><span class="n">kraken</span> <span class="o">-</span><span class="n">o</span> <span class="n">all_samples</span><span class="o">.</span><span class="n">kraken</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
</li>
</ul>
<img alt="krona.png" src="krona.png" />
</div>
<div class="section" id="building-a-kraken-standard-database-only-on-large-clusters">
<h3>Building a Kraken standard database (only on large clusters)<a class="headerlink" href="#building-a-kraken-standard-database-only-on-large-clusters" title="Permalink to this headline">¶</a></h3>
<p>The pre-built Minikraken database is useful for a quick metagenomic screening of shotgun data. However, by building larger databases (i.e. a larger set of k-mers gathered) we may increase the sensitivity of the analysis.
One option is to build the Kraken standard database. To create this database we use the command <em>kraken-build</em>, which downlads the RefSeq complete genomes for the bacterial, archaeal, and viral domains, and builds the database.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">standard</span> <span class="o">--</span><span class="n">db</span> <span class="n">standardkraken</span><span class="o">.</span><span class="n">folder</span>
</pre></div>
</div>
<ul>
<li><p class="first"><em>Note 1</em>: usage of the database will require users to keep only the <strong>database.idx</strong>, <strong>database.kdb</strong>, <strong>taxonomy/nodes.dmp</strong> and <strong>taxonomy/names.dmp</strong> files. During the database building process some intermediate file are created that may be removed afterwards with the command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">db</span> <span class="n">standardkraken</span><span class="o">.</span><span class="n">folder</span> <span class="o">--</span><span class="n">clean</span>
</pre></div>
</div>
</li>
<li><p class="first"><em>Note 2</em>: The downloaded RefSeq genomes require 33GB of disk space. The build process will then require approximately 450GB of additional disk space. The final <strong>database.idx</strong>, <strong>database.kdb</strong>, and <strong>taxonomy/</strong> files reqire 200 Gb of disk space, and running one sample against such database requires 175 Gb of RAM.</p>
</li>
</ul>
</div>
</div>
<div class="section" id="custom-kraken-database">
<h2>Custom Kraken database<a class="headerlink" href="#custom-kraken-database" title="Permalink to this headline">¶</a></h2>
<div class="section" id="building-a-kraken-custom-database-on-hpc-clusters">
<h3>Building a Kraken custom database (on HPC clusters)<a class="headerlink" href="#building-a-kraken-custom-database-on-hpc-clusters" title="Permalink to this headline">¶</a></h3>
<p>Building kraken custum databases is computationally intensive. You will find a ready to use database.
Kraken also allows creation of customized databases, where we can choose which sequences to include and the final size of the database. For example if you do not have the computational resources to build and run analyses with a full database of bacterial genomes (or you don’t need to), you may want to build a custom database with only the genomes needed for your application.</p>
<blockquote>
<div><ol class="arabic">
<li><p class="first">First of all we choose a name for our database and we create a folder with that name using <cite>mkdir</cite>. Let’s call the database <strong>CustomDB</strong>. This will be the name used in all the dollowing commands after the –db option.</p>
</li>
<li><p class="first">Download NCBI taxonomy files (the sequence ID to taxon map, the taxonomic names and tree information) with <strong>kraken-build –download-taxonomy</strong>. The taxonomy files are necessary to associate a taxon to the sequence identifier (the GI number in NCBI) of the fasta sequences composing our database. For this reason we will build our database only with sequences from the NCBI RefSeq. For more information on NCBI taxonomy visit <a class="reference external" href="https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi">https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi</a>. This command will create a sub-folder <strong>taxonomy/</strong> inside our CustomDB folder:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">download</span><span class="o">-</span><span class="n">taxonomy</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span>
</pre></div>
</div>
</li>
<li><p class="first">Install a genomic library. RefSeq genomes in fasta file from five standard groups are made easily available in Kraken with the command <strong>kraken-build –download-library</strong>:</p>
</li>
</ol>
<blockquote>
<div><ul class="simple">
<li><em>bacteria</em> : RefSeq complete bacterial genomes</li>
<li><em>archaea</em> : RefSeq complete archaeal genomes</li>
<li><em>plasmid</em> : RefSeq plasmid sequences</li>
<li><em>viral</em> : RefSeq complete viral genomes</li>
<li><em>human</em> : GRCh38 human genome</li>
</ul>
<p>The following command will download all the RefSeq bacterial genomes (33Gb size) and create a folder <strong>library/</strong> with a sub-folder <strong>bacteria/</strong> inside your CustomDB folder:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">download</span><span class="o">-</span><span class="n">library</span> <span class="n">bacteria</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span>
</pre></div>
</div>
</div></blockquote>
<ol class="arabic" start="4">
<li><p class="first">We can add any sort of RefSeq fasta sequences to the library with <strong>kraken-build –add-to-library</strong>. For example we will add to the library of bacterial genomes the RefSeq sequences of mitochodrial genomes. The sequences will be inside the sub-folder <strong>added/</strong>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>    <span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">add</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">library</span> <span class="n">mitochondrion</span><span class="o">.</span><span class="mf">1.1</span><span class="o">.</span><span class="n">genomic</span><span class="o">.</span><span class="n">fna</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span>
<span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">add</span><span class="o">-</span><span class="n">to</span><span class="o">-</span><span class="n">library</span> <span class="n">mitochondrion</span><span class="o">.</span><span class="mf">2.1</span><span class="o">.</span><span class="n">genomic</span><span class="o">.</span><span class="n">fna</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span>
</pre></div>
</div>
<ul>
<li><p class="first"><em>Note</em>: if you have several fasta files to add you can use a <cite>for</cite> loop:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for i in *.fasta
do
  kraken-build --add-to-library $i --threads 4 --db CustomDB
done
</pre></div>
</div>
</li>
</ul>
</li>
<li><p class="first">Once gathered all the genomes that we want to use for our custom database, low-complexity regions have to be dusted. The program <strong>dustmasker</strong> (from Blast+, more info at <a class="reference external" href="https://www.ncbi.nlm.nih.gov/books/NBK279681/">https://www.ncbi.nlm.nih.gov/books/NBK279681/</a>) identifies low-complexity regions and soft-mask them (the corresponding sequence is turned to lower-case letters). With a <cite>for</cite> loop we run dustmasker on each fasta file present in the library folder, and we will pipe (|) to dustmasker a <cite>sed</cite> command to replace the low-complexity regions (lower-case) with Ns. Notice that the output is redirected (&gt;) to a temporary file, which is afterwards renamed to replace the original file fasta file with the command <cite>mv</cite>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for i in `find CustomDB/library \( -name &#39;*.fna&#39; -o -name &#39;*.ffn&#39; \)`
do
      dustmasker -in $i -infmt fasta -outfmt fasta | sed -e &#39;/&gt;/!s/a\|c\|g\|t/N/g&#39; &gt; tempfile
      mv -f tempfile $i
done
</pre></div>
</div>
</li>
</ol>
<blockquote>
<div><p>Some of the options available in Dustmasker:</p>
<table border="1" class="docutils">
<colgroup>
<col width="42%" />
<col width="58%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Option</th>
<th class="head">Function</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td><em>-in</em> &lt;string&gt;</td>
<td>input file name</td>
</tr>
<tr class="row-odd"><td><em>-infmt</em> &lt;string&gt;</td>
<td>input format (e.g. fasta)</td>
</tr>
<tr class="row-even"><td><em>-outfmt</em> &lt;string&gt;</td>
<td>output format (fasta)</td>
</tr>
</tbody>
</table>
</div></blockquote>
<ol class="arabic" start="6">
<li><p class="first">Finally, we build the database with kraken-build. With this command, Kraken uses all the hard-masked genomes contained in the library (bacteria and mtDNA RefSeq) to create a database of 31 bp-long k-mers. We can choose the size of our custom database (hence the number of k-mers included, and the sensitivity) with the  <cite>–max-db-size</cite> option (8 Gb here).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span><span class="o">-</span><span class="n">build</span> <span class="o">--</span><span class="n">build</span> <span class="o">--</span><span class="nb">max</span><span class="o">-</span><span class="n">db</span><span class="o">-</span><span class="n">size</span> <span class="mi">8</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span>
</pre></div>
</div>
</li>
</ol>
</div></blockquote>
</div>
<div class="section" id="taxonomic-assignation-with-kraken-custom-database">
<h3>Taxonomic assignation with Kraken custom database<a class="headerlink" href="#taxonomic-assignation-with-kraken-custom-database" title="Permalink to this headline">¶</a></h3>
<p>Once our custom database is built we can run the command for taxonomic assignation of DNA reads agaisnt the custom database, as in section 1.1 and 1.2.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">kraken</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span> <span class="o">--</span><span class="n">threads</span> <span class="mi">4</span> <span class="o">--</span><span class="n">fastq</span><span class="o">-</span><span class="nb">input</span> <span class="n">merged</span><span class="o">.</span><span class="n">fastq</span> <span class="o">--</span><span class="n">output</span> <span class="n">sample</span><span class="o">.</span><span class="n">kraken</span>
<span class="n">kraken</span><span class="o">-</span><span class="n">report</span> <span class="o">--</span><span class="n">db</span> <span class="n">CustomDB</span> <span class="n">sample</span><span class="o">.</span><span class="n">kraken</span> <span class="o">&gt;</span> <span class="n">sample</span><span class="o">.</span><span class="n">kraken</span><span class="o">.</span><span class="n">report</span>
</pre></div>
</div>
<p>Or, again, we can loop the commands if we have various samples.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>for i in *.fastq
do
  filename=$(basename &quot;$i&quot;)
  fname=&quot;${filename%.fastq}&quot;
  kraken --db CustomDB --threads 4 --fastq-input $i --output ${fname}.kraken
  kraken-report --db CustomDB ${fname}.kraken &gt; ${fname}.kraken.report
done
</pre></div>
</div>
<p>Finally, we can visualize the results of the Kraken analysis with <em>Krona</em>. Run <em>ktImportTaxonomy</em> to generate the html file and open it in a web browser to notice the difference with the analysis done with Minikraken.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">ktImportTaxonomy</span> <span class="o">-</span><span class="n">q</span> <span class="mi">2</span> <span class="o">-</span><span class="n">t</span> <span class="mi">3</span> <span class="n">sample</span><span class="o">.</span><span class="n">kraken</span> <span class="o">-</span><span class="n">o</span> <span class="n">sample</span><span class="o">.</span><span class="n">kraken</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
</div>
</div>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Amine Namouchi &amp; Claudio Ottoni

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
    <!-- Theme Analytics -->
    <script>
    (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
      (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
      m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
    })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

    ga('create', 'UA-XXXXXXX-1', 'auto');
    ga('send', 'pageview');
    </script>

    
   

</body>
</html>